src/java/fr/paris/lutece/util/signrequest/IPAuthentificator.java

```diff
@@ -0,0 +1,90 @@
+package fr.paris.lutece.util.signrequest;
+
+import java.util.List;
+import java.util.Set;
+
+import javax.servlet.http.HttpServletRequest;
+
+public class IPAuthentificator implements RequestAuthenticator {
+	
+	enum CASE {
+		ALLOW 
+		{
+			public boolean getAuthorisation( ) {
+				return true;
+			}
+		},
+		BLOCK
+		{
+			public boolean getAuthorisation( ) {
+				return false;
+			}
+		};
+		
+		public abstract boolean getAuthorisation( );
```

Comment:
I think it would be more clear to have a simple enum here, and to put the logic in isRequestAuthenticated with a switch statement for instance.